### PR TITLE
Add erorr handling when stock to create already exists

### LINKED ISSUE #14593: `productVariantBulkUpdate` error policy does not properly reject failed rows
Using the `productVariantBulkUpdate` mutation to create stocks does not allow for the error policy to gracefully handle failed errors. Instead, Saleor throws a 500 server error. This happens if you try to call create on the same variant warehouse stock more than once.
```
mutation updateChannel($id: ID!, $variants: [ProductVariantBulkUpdateInput!]!) {
        productVariantBulkUpdate(errorPolicy: REJECT_FAILED_ROWS, product: $id, variants: $variants) {
            errors {
                field
                message
            }
            results {
                errors {
                    message
                }
                productVariant {
                    id
                }
            }
        }
    }
```
This mutation with the following input
```
{
  "id": "UHJvZHVjdDo2MQ==",
  "variants": [
    {
      "id": "UHJvZHVjdFZhcmlhbnQ6MTcwNzk=",
      "stocks": {
        "create": [
          {
            "warehouse": "V2FyZWhvdXNlOjE1YzRmM2E1LTQwNGEtNDg0Zi1iMjY0LThjYjdhOTE1N2U3OA==",
            "quantity": 100
          },
          {
            "warehouse": "V2FyZWhvdXNlOjIyZTJlODQ3LTgzNWMtNDcwMi1hMDJjLTI1YjhiNWZlNWEzMQ==",
            "quantity": 100
          },
          {
            "warehouse": "V2FyZWhvdXNlOjdlZTE5Nzc0LWJiZTQtNDA5YS1iMjBiLTg4YWNlOTA2Zjg1NA==",
            "quantity": 100
          },
          {
            "warehouse": "V2FyZWhvdXNlOmQ0ZGNhZDc5LWI0MmEtNGRhZC04M2VhLTdkMDEyYzRjZjkwYQ==",
            "quantity": 100
          }
        ]
      }
    }
  ]
}
```
Will succeed if the stock has never been created, but when running it again it throws a server error.  There are instances where we do not know if the stock has been created, so we attempt it again as there is no 'upsert' function provided in the schema and would like to simply ignore the errors.

### PR DESCRIPTION
**Pull request by SzymJ**
I want to merge this change because it fixes #14593

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py [hunk 1]
```diff
@@ -410,6 +433,23 @@ def clean_variant(
 
         return cleaned_input if cleaned_input else None
 
+    @classmethod
+    def _get_input_warehouses_its(cls, variants_data):
```
Comment (IKarbowiak) [inline saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py:437]: Looks like typo in function name. Shouldn't be `_get_input_warehouses_ids`?

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py [hunk 2]
```diff
@@ -289,7 +289,30 @@ def clean_stocks(
...
+            warehouse_global_id = graphene.Node.to_global_id(
+                "Warehouse", stock.warehouse_id
+            )
+            used_warehouses[warehouse_global_id].append(stock.product_variant_id)
+
         if stocks_data := cleaned_input["stocks"].get("create"):
+            for stock_index, stock in enumerate(stocks_data):
+                if (
+                    stock["warehouse"] in used_warehouses
+                    and cleaned_input["id"].id in used_warehouses[stock["warehouse"]]
```
Comment (maarcingebala) [inline saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py:303]: Is it `cleaned_input["id"].id` and not `cleaned_input["id"]`?

Comment (korycins) [reply saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py:303]: I might be wrong, but I think you can simplify the condition:
```python
if cleaned_input["id"] in used_warehouses.get(stock["warehouse"], {}):
```

**Review by korycins — APPROVED** [review]

File: saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py (hunk 2 above)
Comment (maarcingebala) [reply saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py]: @SzymJ Btw if tests didn't catch the `cleaned_input["id"].id` issue I assume some paths must have been not covered 🤔

**Review by szczecha — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**